Web3-Validator build error: Type '"type"' does not satisfy the constraint

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When using `web3-validator` package in other project which have a dependency to `ajv` turns into following build errors. 

```
$ tsc
../../node_modules/web3-validator/dist/types.d.ts:18:68 - error TS2344: Type '"type"' does not satisfy the constraint 'number | unique symbol | "length" | "toString" | "charAt" | "charCodeAt" | "concat" | "indexOf" | "lastIndexOf" | "localeCompare" | "match" | "replace" | "search" | "slice" | ... 33 more ... | "trimRight"'.

18 export declare type JsonSchema = Optional<JSONSchemaType<unknown>, "type"> & {
```

### Expected Behavior

The build should be fine. 

### Steps to Reproduce

- Add `web3-validator` as dependency to a TS project. 
- Add the `ajv` as direct dependency to that project. 
- Try building the project. 

### Web3.js Version

4.0.1-alpha.2

### Environment

_No response_

### Anything Else?

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -945,3 +945,9 @@ should use 4.0.1-alpha.0 for testing.
 -   These types were moved from `web3-eth-accounts` to `web3-types` package: Cipher, CipherOptions, ScryptParams, PBKDF2SHA256Params, KeyStore (#5581 )
 
 ## [Unreleased]
+
+### Fixed
+
+#### web3-validator
+
+-   Fix build error '"type"' does not satisfy the constraint (#5712)

--- a/packages/web3-validator/src/types.ts
+++ b/packages/web3-validator/src/types.ts
@@ -16,7 +16,8 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { AbiParameter, Optional } from 'web3-types';
-import { ErrorObject, JSONSchemaType } from 'ajv';
+import { ErrorObject } from 'ajv';
+import { SomeJSONSchema } from 'ajv/dist/types/json-schema';
 
 export { JSONSchemaType } from 'ajv';
 export { DataValidateFunction, DataValidationCxt } from 'ajv/dist/types';
@@ -64,6 +65,6 @@ export type Web3ValidationOptions = {
 
 // In `JSONSchemaType` from `ajv` the `type` is required
 // We need to make it optional
-export type JsonSchema = Optional<JSONSchemaType<unknown>, 'type'> & {
+export type JsonSchema = Optional<SomeJSONSchema, 'type'> & {
 	readonly eth?: string;
 };

